frontend/src/hooks/useLocations.js

```diff
@@ -2,15 +2,16 @@ import {useEffect, useState} from "react";
 import axios from "axios";
 
 
-export default function useLocations() {
+export default function useLocations(geoLocation) {
 
     const [locations, setLocations] = useState([])
+    const url = '/api/location' + [geoLocation? ["?lat=" + geoLocation.latitude + "&lng=" + geoLocation.longitude] : ""]
```

Comment:
Why the square brackets?

Maybe use a template string instead of concatenation?